script.js

```diff
@@ -1,4 +1,420 @@
+  var index = 0;
+  while (index < playerHandArray.length) {
+    playerMessage =
+      playerMessage +
+      " - " +
+      playerHandArray[index].name +
+      " of " +
+      playerHandArray[index].suit +
+      "<br>";
+    index = index + 1;
+  }
+  // DEALER
+  var dealerMessage = "Dealer Hand:<br>";
+  var index = 0;
+  while (index < dealerHandArray.length) {
+    dealerMessage =
+      dealerMessage +
+      " - " +
+      dealerHandArray[index].name +
+      " of " +
+      dealerHandArray[index].suit +
+      "<br>";
+    index = index + 1;
+  }
+  return playerMessage + "<br>" + dealerMessage;
+};
+
+// ==== show total value of player & dealer hands ====
+var showHandTotalValues = function (playerHandValue, dealerHandValue) {
+  var totalHandValueMessage =
+    "<br>Player total hand value: " +
+    playerHandValue +
+    "<br>Dealer total hand value:" +
+    dealerHandValue;
+  return totalHandValueMessage;
+};
+
+// ==== Function calclate total hand value ====
+var totalHandValue = function (handArray) {
+  var totalHandValue = 0;
+  var aceCounter = 0;
+  // loop through player & com hand and add values
+  var index = 0;
+  while (index < handArray.length) {
+    var currentCard = handArray[index];
+    if (
+      currentCard.name == "jack" ||
+      currentCard.name == "queen" ||
+      currentCard.name == "king"
+    ) {
+      totalHandValue = totalHandValue + 10;
+    } else if (currentCard.name == "ace") {
+      totalHandValue = totalHandValue + 11;
+      aceCounter = +1;
+    } else {
+      totalHandValue = totalHandValue + currentCard.rank;
+    }
+    index = index + 1;
+    while (index < aceCounter) {
+      totalHandValue = totalHandValue - 10;
+    }
+    index = index + 1;
+  }
+
+  return totalHandValue;
+};
+
+// ==== Function for shuffled deck ====
+var shuffledDeck = function () {
+  var newDeck = makeDeck();
+  var shuffledDeck = shuffleCards(newDeck);
+  return shuffledDeck;
+};
+
+// ==== function to reset game ====
+var reset = function () {
+  currentMode = gameStart;
+  gameDeck = "empty at the start";
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // 1st click - click to shuffle cards
+  var message = "";
+  if (currentGameMode == gameStart) {
+    gameDeck = shuffledDeck();
+
+    playerHand.push(gameDeck.pop());
+    playerHand.push(gameDeck.pop());
+    computerHand.push(gameDeck.pop());
+    computerHand.push(gameDeck.pop());
+
+    currentGameMode = drawCards;
+    // Approach through little steps first e.g. shuffle first etc
+    message = "Ya'll been dealt cards bruh";
+
+    return message;
+  }
+  // 2nd click - check for blackjack
+  if (currentGameMode == drawCards) {
+    var playerBlackjack = BlackjackCheck(playerHand);
+    var comBlackJack = BlackjackCheck(computerHand);
+
+    //  player & com have blackjack = draw
+    //  boolean false add ! in front of var
+    if (playerBlackjack || comBlackJack) {
```

Comment:
The whole point of this condition is just to get a message for based on the player vs computer hand. Consider refactoring this whole chunk into a helper function to reduce the length of your main function